### PR TITLE
LSTCore development

### PR DESCRIPTION
**Pull request by ariostas**
This is a continuation of #19, but now using CMSSW_14_1_0_pre3. Now LSTCore uses the master TrackLooper branch (thanks to https://github.com/SegmentLinking/TrackLooper/pull/395), and both LST and LSTCore packages are compiled for all backends (thanks to https://github.com/SegmentLinking/TrackLooper/pull/385 and the cmssw-config fixes).

### CODE REVIEW

File: RecoTracker/LST/plugins/alpaka/LSTModulesDevESProducer.cc [hunk 1]
```diff
@@ -8,9 +8,8 @@
 #include "RecoTracker/Record/interface/TrackerRecoGeometryRecord.h"
 
 // LST includes
-#include <SDL/Module.h>
-#include <SDL/ModuleMethods.h>
-#include <SDL/LST.h>
+#include <RecoTracker/LSTCore/interface/alpaka/Module.h>
+#include <RecoTracker/LSTCore/interface/alpaka/LST.h>
```
Comment (slava77) [inline RecoTracker/LST/plugins/alpaka/LSTModulesDevESProducer.cc:12]:
```suggestion
#include "RecoTracker/LSTCore/interface/alpaka/LST.h"
```

File: RecoTracker/LSTCore/BuildFile.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<use name="alpaka"/>
+<use name="boost_header"/>
+<use name="root"/>
+<use name="HeterogeneousCore/AlpakaInterface"/>
+<flags CXXFLAGS="-DLST_IS_CMSSW_PACKAGE -DCACHE_ALLOC -DT4FromT3 -DUSE_RZCHI2 -DUSE_T5_DNN -DPT_CUT=0.8 -DDUP_pLS -DDUP_T5 -DDUP_pT5 -DDUP_pT3 -DCrossclean_T5 -DCrossclean_pT3"/>
```
Comment (slava77) [inline RecoTracker/LSTCore/BuildFile.xml:5]:
```suggestion
<flags CXXFLAGS="-DLST_IS_CMSSW_PACKAGE -DCACHE_ALLOC -DT4FromT3 -DUSE_RZCHI2 -DUSE_T5_DNN -DPT_CUT=0.8 -DDUP_pLS -DDUP_T5 -DDUP_pT5 -DDUP_pT3 -DCrossclean_T5 -DCrossclean_pT3 -Wshadow"/>
```

let's keep `Wshadow` for now, similar to what we have in the standalone/non-CMSSW build.

**Comment by VourMa** [comment]
@ariostas What is the plan here?
In my mind, we would close it and keep the `_LSTCore` branch with the submodule for now. Then, we would create a new branch, say `_LSTCoreProper`, with actual files instead of the submodule. and that's what we merge to `CMSSW_14_1_0_pre3_LST_X`. Or is it the plan to make the `_LSTCore` branch have the actual files?

**Comment by ariostas** [comment]
@VourMa yeah, that sounds good. I'm just updating things here so I can test that everything is working properly.

**Pull request closed without merging**